pymc/aesaraf.py

```diff
@@ -145,6 +145,7 @@ def convert_observed_data(data):
 def change_rv_size(
     rv: TensorVariable,
     new_size: PotentialShapeType,
+    new_size_dims: Optional[tuple] = (None,),
```

Comment:
The name info should not be handled by this function. I think somewhere in the model we create constants or shared variables for dims sizes and there is where we should set the names.

Reply:
Could you give me a hint roughly where this should be? Before or after the call to change_rv_size?

Reply:
So, I looked into it and when I set the names in the suggested lines, I receive only the following output from dprint:

```
normal_rv{0, (0, 0), floatX, False}.1 [id A] 'x'
 |RandomGeneratorSharedVariable(<Generator(PCG64) at 0x7F99BD4B6960>) [id B]
 |TensorConstant{(1,) of 5} [id C]
 |TensorConstant{11} [id D]
 |TensorConstant{0} [id E]
 |TensorConstant{1.0} [id F]
```

only after passing the name to new_size in aesaraf.py::change_rv_size, I get the change
https://github.com/pymc-devs/pymc/blob/be048a495b32a09c49ec8d3e8a95d4f5e4290d2a/pymc/aesaraf.py#L185

```python
new_size = at.as_tensor(new_size, ndim=1, dtype="int64", name=new_size[0].name)
```

```
normal_rv{0, (0, 0), floatX, False}.1 [id A] 'x'
 |RandomGeneratorSharedVariable(<Generator(PCG64) at 0x7FAD2928A960>) [id B]
 |cities_dim{(1,) of 5} [id C]
 |TensorConstant{11} [id D]
 |TensorConstant{0} [id E]
 |TensorConstant{1.0} [id F]
```

If this is okay, I'll implement the changes in this PR

Reply:
This PR added a name for the mutable case, but not the constant one: https://github.com/pymc-devs/pymc/pull/6147/commits/f45ca4a20c4868693769f14892bb9aab7a3e93fe

Reply:
Thanks, I will try to give it another go in this week

Reply:
@ricardoV94, I'm dancing around this PR for quite a bit now, but I'm not finding the time to do it. In the beginning of next year I would have more time, but if it bothers you to have an open PR for such a long time or someone else takes over, I won't mind if I don't get to finish it.

Reply:
No worries. The original issue lost some of it's relevance when we switched from defaulting resizable dims to constant dims. It's still nice but not urgent by any means. Feel free to come back to it whenever you find the time.